docs/source-2.0/tutorials/full-stack-tutorial.rst

```diff
@@ -0,0 +1,880 @@
+        },
+        "plugins": {
+            "typescript-ssdk-codegen": {
+                "package" : "@com.example/coffee-service-server",
+                "packageVersion": "0.0.1"
+            }
+        }
+    }
+
+Run the build:
+
+.. code-block:: sh
+    :caption: ``/bin/sh``
+
+    smithy build model/
+
+The build will should fail for the following reason:
+
+.. code-block:: text
+    :caption: ``failure message``
+
+    Projection source failed: software.amazon.smithy.codegen.core.CodegenException:
+        Every operation must have the smithy.framework#ValidationException error attached
+            unless disableDefaultValidation is set to 'true' in the plugin settings.
+        Operations without smithy.framework#ValidationException errors attached:
+            [com.example#CreateOrder, com.example#GetMenu, com.example#GetOrder]
+
+
+The server SDK validates inputs by default, and enforces each operation has
+the ``smithy.framework#ValidationException`` attached to it. We will fix this issue by attaching the error
+to our service, meaning all operations in the service may return it. Let's do this now:
+
+.. code-block:: smithy
+    :caption: ``main.smithy``
+
+    use aws.protocols#restJson1
+    use smithy.framework#ValidationException
+
+    ...
+    service CoffeeShop {
+        ...
+        errors: [
+            ValidationException
+        ]
+    }
+
+
+After fixing this and running the build, the TypeScript code-generator plugin will create a new
+artifact under ``build/smithy/source/typescript-ssdk-codegen``. This artifact contains the generated
+server SDK (SSDK), which we will use in our back-end.
+
+Implementing the server
+=======================
+For this tutorial, we have included a ``Makefile``, which simplifies the process of building and running the
+application. To use it, make sure to run ``make`` from the root of the application directory (where the ``Makefile``
+lives). Let's try it now:
+
+.. code-block:: sh
+    :caption: ``/bin/sh``
+
+    make build-server
+
+This command will run the code-generation for the server SDK, and then build the server implementation (which uses
+the server SDK). The server package is located under the ``server/`` directory, and contains
+only two files under ``src/``:
+
+* ``index.ts``: entry-point of the backend application, and where we initialize our service
+* ``CoffeeShop.ts``: implementation of a `CoffeeShopService` from the generated server SDK
+
+The ``ssdk/`` directory is a link to our generated server SDK, which is an output of the smithy build. This is where
+the server imports the generated code from. Let's take a look at the core of the coffee shop implementation:
+
+.. code-block:: TypeScript
+    :caption: ``CoffeeShop.ts``
+
+    // An implementation of the service from the SSDK
+    export class CoffeeShop implements CoffeeShopService<{}> {
+        ...
+
+        CreateOrder(input: CreateOrderServerInput, context: CoffeeShopContext): CreateOrderServerOutput {
```

Comment:
These methods are out of sync with the template (missing `async`, not returning a promise)